OloEditor/src/EditorLayer.cpp

```
@@ -514,21 +571,47 @@ namespace OloEngine
                         m_HoveredEntity.GetComponent<SpriteRendererComponent>().Texture = texture;
+                        auto newComponent = m_HoveredEntity.GetComponent<SpriteRendererComponent>();
+                        m_CommandHistory.PushAlreadyExecuted(
+                            std::make_unique<ComponentChangeCommand<SpriteRendererComponent>>(
+                                m_EditorScene, m_HoveredEntity.GetUUID(), oldComponent, newComponent));
                     }
                     else
                     {
                         OLO_WARN("Could not load texture {0}", path.filename().string());
                     }
                 }
+                else if (m_SceneState == SceneState::Edit && path.extension() == ".oloprefab") // Instantiate prefab
```

Clang (14.0.6)

[note] +1, nesting level increased to 3

(clang)